doc/DataPlaneCodeGeneration/AzureSDKCodeGeneration_DataPlane_Quickstart.md

```diff
@@ -52,49 +53,102 @@ For this guide, we'll create a getting started project in a branch of your fork
 ```text
 sdk\<service name>\<package name>\README.md
 sdk\<service name>\<package name>\api
+sdk\<service name>\<package name>\perf
+sdk\<service name>\<package name>\samples
 sdk\<service name>\<package name>\src
+sdk\<service name>\<package name>\stress
 sdk\<service name>\<package name>\tests
-sdk\<service name>\<package name>\samples
 sdk\<service name>\<package name>\CHANGELOG.md
+sdk\<service name>\<package name>\<package name>.sln
 ```
 
 - `<service name>` - Should be the short name for the azure service. e.g. deviceupdate
 - `<package name>` -  Should be the name of the shipping package, or an abbreviation that distinguishes the given shipping artifact for the given service. It will be `Azure.<group>.<service>`, e.g. Azure.IoT.DeviceUpdate
+
+### Use Cadl as Input
+
+#### Create Cadl project
   
-We will use dotnet project template [Azure.Template](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/template/Azure.Template) to automatically create the project.
+  ***Initialize Cadl Project***
 
-You can run `eng\scripts\automation\Invoke-DataPlaneGenerateSDKPackage.ps1` to generate the starting SDK client library package directly as following:
+  Follow [Cadl Getting Start](https://github.com/microsoft/cadl/#using-node--npm) to initialize your Cadl project.
 
-```powershell
-eng/scripts/automation/Invoke-DataPlaneGenerateSDKPackage.ps1 -service <servicename> -namespace Azure.<group>.<service> -sdkPath <sdkrepoRootPath> [-inputfiles <inputfilelink>] [-readme <readmeFilelink>] [-securityScope <securityScope>] [-securityHeaderName <securityHeaderName>]
+  Make sure `npx cadl compile .` runs correctly.
+
+  ***Add cadl-csharp emitter***
+
+  Modify package.json, add one line under dependencies:
+
+```diff
+        "dependencies": {
+          "@cadl-lang/compiler": "^0.36.0",
+          "@cadl-lang/rest": "^0.18.0",
+          "@azure-tools/cadl-azure-core": "^0.8.0",
++         "@azure-tools/cadl-csharp": "0.1.6"
+        },
 ```
 
-e.g.
+        Run `npm install` again to install @azure-tools/cadl-csharp.
+
+  ***Modify (or create) cadl-project.yaml, add one line under emitters:***
+
+  ```diff
+  emitters:
++  "@azure-tools/cadl-csharp": true
+  ```
+
+  ***Configuration (optional)***
+
+  One can further configure the SDK generated, using the emitter options on @azure-tools/cadl-csharp.
 
-```powershell
-pwsh /home/azure-sdk-for-net/eng/scripts/automation/Invoke-DataPlaneGenerateSDKPackage.ps1 -service webpubsub -namespace Azure.Messaging.WebPubSub -sdkPath /home/azure-sdk-for-net -inputfiles https://github.com/Azure/azure-rest-api-specs/blob/73a0fa453a93bdbe8885f87b9e4e9fef4f0452d0/specification/webpubsub/data-plane/WebPubSub/stable/2021-10-01/webpubsub.json -securityScope https://sample/.default
+```
+emitters:
+  "@azure-tools/cadl-csharp":
```

Comment:
Can we provide a list of all properties supported?

Reply:
updated